Development/cpprest/http_utils.cpp

```diff
@@ -330,6 +330,111 @@ namespace web
                 return result;
             }
 
+            utility::string_t make_directives_header(const directives& values)
+            {
+                utility::string_t result;
+                for (auto& value : values)
+                {
+                    if (!result.empty()) { result.push_back(U(';')); }
+                    result.append(value.first);
+                    if (!value.second.empty())
+                    {
+                        result.push_back(U('='));
+                        result.append(value.second);
+                    }
+                }
+                return result;
+            }
+
+            directives parse_directives_header(const utility::string_t& value)
+            {
+                enum {
+                    pre_param,
+                    pre_param_name,
+                    param_name,
+                    pre_param_value,
+                    param_value,
+                    param_value_token,
+                    param_value_quoted_string,
+                    param_value_quoted_string_escape
+                } state = pre_param_name;
+
+                // token      = 1*<any CHAR except CTLs or separators>
+                // separators = "(" | ")" | "<" | ">" | "@"
+                //            | "," | ";" | ":" | "\" | <">
+                //            | "/" | "[" | "]" | "?" | "="
+                //            | "{" | "}" | SP | HT
+                // see https://tools.ietf.org/html/rfc2616#section-2.2
+                auto is_tchar = [](utility::char_t c)
```

Comment:
Since RFC 2616 is obsoleted by RFC 7230, I checked the definition of `token` and `tchar` there. Although it's expressed differently,  the result is the equivalent. So maybe you could get rid of this whole comment and function and reuse the `is_tchar` function above?